// gor-react-accordion/lib/Accordion.js
'use strict';

var React = require('react');
var Panel = require('./Panel');
var Accordion = React.createClass({
  displayName: 'Accordion',


  getInitialState: function getInitialState(panelList) {
    return { openPanelIndex: 0 };
  },
  buildSections: function buildSections(panelList) {
    var sections = panelList.map(this.buildSection);
    return sections;
  },
  buildSection: function buildSection(panel, index) {
    var openStatus = index === this.state.openPanelIndex;
    /* Remember to add a 'key'. React wants you to add an identifier when you instantiate a component multiple times */
    return React.createElement(Panel, { key: index, idx: index, data: panel, toggleOne: this.toggleOne, open: openStatus });
  },
  toggleOne: function toggleOne(id) {
    if (this.state.openSectionIndex === id) {
      this.setState({ openPanelIndex: -1 });
    } else {
      this.setState({ openPanelIndex: id });
    }
  },
  render: function render() {
    var panels = this.buildSections(this.props.data);
    return React.createElement(
      'div',
      { className: 'accordion' },
      panels
    );
  }
});
/*Accordion.propTypes = {
  "data":React.propTypes.object.isRequired
};*/
module.exports = Accordion;
